setup.py

```diff
@@ -36,6 +40,7 @@
     ],
     platforms=['Linux'],
     install_requires=requirements,
+    python_requires='>=3.5.2',
```

Comment:
If it is python 3.5.2 here, shouldn't it be python 3.5.2 everywhere?

Reply:
That's in here because of the typing.type we use somewhere (don't remember exactly). But I am not sure whether we wanted to keep that or change the line to be compatible with 3.5.*?

Reply:
I don't think there's a reason why anyone should use python>=3.5.0 && python<3.5.2 so we can be strict here.